### PR TITLE
refactor: Optimize code based on cargo clippy suggestions 

### PR DESCRIPTION
**Pull request by InventiveCoder**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3415,11 +3415,11 @@ impl Build {
             })
             .map(|prefix| *prefix)
             .or_else(||
-            // If no toolchain was found, provide the first toolchain that was passed in.
-            // This toolchain has been shown not to exist, however it will appear in the
-            // error that is shown to the user which should make it easier to search for
-            // where it should be obtained.
-            prefixes.first().map(|prefix| *prefix))
+                // If no toolchain was found, provide the first toolchain that was passed in.
```
Comment (NobodyXu) [inline src/lib.rs:3418]: I'd recommend moving the comment outside the closure

File: src/lib.rs [hunk 2]
```diff
@@ -236,14 +236,17 @@ use std::sync::{Arc, Mutex};
 #[cfg(feature = "parallel")]
 mod parallel;
 mod windows;
+
 // Regardless of whether this should be in this crate's public API,
 // it has been since 2015, so don't break it.
 pub use windows::find_tools as windows_registry;
 
 mod command_helpers;
+
```
Comment (NobodyXu) [inline src/lib.rs:245]: I personally think it's ok for `use` to on the next line of `mod`

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (InventiveCoder) [reply src/lib.rs]: Thanks!

Modified and execute cargo fmt.

Please review again.

**Review by NobodyXu — APPROVED** [review]
Thanks!

**Pull request merged**